carrot_cli/src/carrot_cli/helper/carrot_helper.py

```diff
@@ -0,0 +1,901 @@
+          "`master`, or the name of your feature branch.")
+    branch = None
+    while not branch or not is_url_accessible(f"{repo}/tree/{branch}"):
+        branch = input("\tPlease enter branch name: ")
+    configs[CONF_BRANCH] = branch
+
+    womtool_path = None
+    while not womtool_path or not is_valid_womtool(womtool_path):
+        womtool_path = input("\nPlease enter the path to womtool: ")
+    configs[CONF_WOMTOOL] = womtool_path
+
+    CarrotHelper.persist_object(configs, CONFIGS_JSON)
+    return configs
+
+
+def get_config():
+    try:
+        with open(CONFIGS_JSON, "r") as f:
+            config = json.loads(f.read())
+            if CONF_BRANCH not in config or \
+                    CONF_REPO not in config or \
+                    CONF_WOMTOOL not in config:
+                raise
+            else:
+                return config
+    except FileNotFoundError:
+        return init_config()
+    except (Exception, json.decoder.JSONDecodeError):
+        print(f"The configurations JSON file is corrupt. Please check if the "
+              f"file `{CONFIGS_JSON}` is a valid JSON and contains the "
+              f"following keys: `{CONF_REPO}`, `{CONF_BRANCH}`, and "
+              f"`{CONF_WOMTOOL}`. Please fix the file and re-try. "
+              f"Alternatively, you may delete the file and re-try, which "
+              f"will trigger re-initialization.")
+        exit()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog=sys.argv[0],
+        description="Helper methods to create, setup, and run carrot "
+                    "tests. Currently, we only support invoking this script "
+                    "from the `wdl_test` directory.")
+    subparsers = parser.add_subparsers(dest="commands", help="Commands")
+
+    subparsers.add_parser(
+        "config",
+        help="Configures communication with the carrot server for defining, "
+             "running, and tracking the tests by asking the user for the "
+             "github repository and the branch where the tests are available "
+             "from, and the path to a local installation of womtool.")
+
+    test_parser = subparsers.add_parser(
+        "test", help="Create, list, and run tests.")
+    test_subparsers = test_parser.add_subparsers(
+        title="sub-commands for test",
+        dest="test", help="Commands to interact with carrot tests.")
+    test_run_parser = test_subparsers.add_parser("run")
+    test_run_parser.add_argument("tests_dir", nargs="+")
+    test_subparsers.add_parser("update_status")
+
+    subparsers.add_parser(
+        "prune",
+        help="Deletes Carrot pipeline and template created by the user. "
+             "Note that Carrot does not currently deletes resources "
+             "belonging to successful executions.")
+
+    args = parser.parse_args()
+
+    # Currently we do not support invoking this script
+    # from a directory other than the wdl_test dir.
```

Comment:
Would it be possible to add an option to the commands (like `--wdl_dir` or something) that allows specifying a directory to use as the wdl_test dir, and just default to the current dir?